pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ReaderTest.java

```diff
@@ -77,7 +78,7 @@ public class ReaderTest extends MockedPulsarServiceBaseTest {
 
     private static final String subscription = "reader-sub";
 
-    @BeforeMethod
+    @BeforeClass
```

Comment:
```suggestion
    @BeforeClass(alwaysRun = true)
```
I'd recommend always adding `(alwaysRun = true)` also for BeforeClass since there's some surprising behavior if test methods are applied to different test groups later.